src/compiler/utilitiesPublic.ts

```diff
@@ -269,7 +269,7 @@ namespace ts {
         return false;
     }
 
-    export function isEmptyBindingElement(node: BindingElement): boolean {
+    export function isEmptyBindingElement(node: BindingElement | ArrayBindingElement): boolean {

```

Comment:
Ugh, this is frustrating. There should be `ObjectBindingElement` and `BindingElement`.

Reply:
Yes, there totally should. I had a really, really hard time understanding the relationship between all of the binding types.